### PR TITLE
Support SubnetRentalRequest NNS function 52

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

There is a new proposal type to rent subnets, using NNS function 52.

# Changes

1. Copy Rust types `SubnetRentalRequest` and `RentalConditionId` from the IC repo to `rs/proposals/src/def.rs`.
2. Render the payload as json without transformation.

# Tests

1. Added a test in `scripts/nns-dapp/test-proposal-payload`.
2. Use `--test-neuron-proposer` because `ic-admin` doesn't work for subnet rental with `--secret-key-pem`.
3. Update snsdemo to a release that supports subnet rentals.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: rs/proposals/src/def.rs [hunk 1]
```diff
@@ -492,3 +492,16 @@ pub type ReviseElectedHostosVersionsPayload = crate::canisters::nns_registry::ap
 // NNS function 51 - DeployHostosToSomeNodes
 // https://github.com/dfinity/ic/blob/26098e18ddd64ab50d3f3725f50c7f369cd3f90e/rs/registry/canister/src/mutations/do_update_nodes_hostos_version.rs#L38C12-L38C43
 pub type DeployHostosToSomeNodesPayload = crate::canisters::nns_registry::api::UpdateNodesHostosVersionPayload;
+
+// Copied from https://github.com/dfinity/ic/blob/master/rs/nns/governance/src/governance.rs
+#[derive(candid::CandidType, candid::Deserialize, serde::Serialize, Clone, Debug)]
```
Comment (mstrasinskis) [inline rs/proposals/src/def.rs:497]: What do you think about having no `Debug` to decrease a binary a bit?

File: scripts/nns-dapp/test-proposal-payload [hunk 2]
```diff
@@ -177,6 +178,20 @@ test_nns_function_51() {
   verify_payload "$subcommand" "$ACTUAL_PAYLOAD" "$EXPECTED_PAYLOAD"
 }
 
+test_nns_function_52() {
+  subcommand="propose-to-rent-subnet"
+
+  PROPOSAL_ID="$(run_ic_admin \
+    "$subcommand" \
+     --rental-condition-id App13CH \
```
Comment (mstrasinskis) [inline scripts/nns-dapp/test-proposal-payload:186]: Is `App13CH` a string?

Comment (mstrasinskis) [reply scripts/nns-dapp/test-proposal-payload:186]: Ah, it's an enum. Thanks got it.

File: rs/proposals/src/def.rs (hunk 1 above)
Comment (dskloetd) [reply rs/proposals/src/def.rs:497]: I don't think it uses space if it's not used.
I built with and without the `Debug` trait and the WASM binaries are identical.

**Comment by mraszyk** [comment]
Does this also address the proposal appearing on the SNS Launchpad?

**Comment by dskloetd** [comment]
Why would the proposal appear on the SNS Launchpad?

File: rs/proposals/src/def.rs (hunk 1 above)
Comment (mstrasinskis) [reply rs/proposals/src/def.rs:497]: Good to know. Thanks for testing 👍

**Review by mstrasinskis — APPROVED** [review]

**Review by mstrasinskis — COMMENTED** [review]
LTGM, thanks!

**Pull request merged**